### PR TITLE
docs: update for standard-readme compliance

### PR DESCRIPTION
**Pull request by thephez**
## Issue being fixed or feature implemented

Brings readme into [standard-readme](https://github.com/RichardLitt/standard-readme) compliance

## What was done?

- [x] Add ToC and make minor formatting/wording adjustments
- [x] Update code example
- [x] Update badges (add release date, change package version -> npm version)

## How Has This Been Tested?

N/A

## Breaking Changes

N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,21 @@
 # Dash SDK
 
-[![Package Version](https://img.shields.io/github/package-json/v/dashevo/dashjs.svg?&style=flat-square)](https://www.npmjs.org/package/dash)
+[![NPM Version](https://img.shields.io/npm/v/dash)](https://img.shields.io/npm/v/dash)
```
Comment (strophy) [inline README.md:3]: Actually shouldn't these badges still link to the NPM repo in case they are clicked?

File: README.md [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 [![Build Status](https://img.shields.io/travis/com/dashevo/dashjs.svg?branch=master&style=flat-square)](https://travis-ci.com/dashevo/dashjs)
+[![Release Date](https://img.shields.io/github/release-date/dashevo/dashjs)](https://img.shields.io/github/release-date/dashevo/dashjs)
+[![standard-readme compliant](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
```
Comment (strophy) [inline README.md:6]: Also can we consistently use either the default style `flat` or the `flat-square` style for all badges?

File: README.md (hunk 1 above)
Comment (dalibran) [reply README.md:3]: that would be my expectation as a user

Comment (thephez) [reply README.md:3]: Yep, that was an accidental change 🙈

File: README.md (hunk 2 above)
Comment (thephez) [reply README.md:6]: Yes, let's do that. Do you have a preference?

Comment (dalibran) [reply README.md:6]: I don't have one, dealer's choice unless @strophy has a strong opinion

Comment (strophy) [reply README.md:6]: I prefer `flat` because it is the default. So we can just omit any `style` specification.

File: README.md (hunk 1 above)
Comment (thephez) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (thephez) [reply README.md]: Fixed

**Pull request merged**